internal/metrics/client_test.go

```diff
@@ -0,0 +1,665 @@
+		metrics             []parser.Metric
+		mockResponse        QueryResponse
+		mockStatus          int
+		expectedMetricKey   string
+		expectedMetricValue float64
+		expectedError       string
+	}{
+		{
+			name: "successful RPS query",
+			metrics: []parser.Metric{
+				{Name: "rps"},
+			},
+			mockResponse: QueryResponse{
+				Status: "success",
+				Data: struct {
+					ResultType string `json:"resultType"`
+					Result     []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					} `json:"result"`
+				}{
+					ResultType: "vector",
+					Result: []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					}{
+						{
+							Value: []interface{}{1234567890, "42.5"},
+						},
+					},
+				},
+			},
+			mockStatus:          http.StatusOK,
+			expectedMetricKey:   "rps",
+			expectedMetricValue: 43, // Rounded up from 42.5
+		},
+		{
+			name: "custom metric",
+			metrics: []parser.Metric{
+				{Name: "custom_metric"},
+			},
+			mockResponse: QueryResponse{
+				Status: "success",
+				Data: struct {
+					ResultType string `json:"resultType"`
+					Result     []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					} `json:"result"`
+				}{
+					ResultType: "vector",
+					Result: []struct {
+						Metric map[string]string `json:"metric"`
+						Value  []interface{}     `json:"value"`
+						Values [][]interface{}   `json:"values"`
+					}{
+						{
+							Value: []interface{}{1234567890, "42.5"},
+						},
+					},
+				},
+			},
+			mockStatus:          http.StatusOK,
+			expectedMetricKey:   "custom_metric",
+			expectedMetricValue: 43, // Rounded up from 42.5
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(tt.mockStatus)
+				json.NewEncoder(w).Encode(tt.mockResponse)
+			}))
+			defer server.Close()
+
+			client := NewMetricsClient(server.URL)
+			responses, err := client.GetMetrics(context.Background(), tt.metrics)
+
+			if tt.expectedError != "" {
+				if err == nil || err.Error() != tt.expectedError {
+					t.Errorf("expected error %q, got %v", tt.expectedError, err)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+
+			if len(responses) == 0 {
+				t.Error("expected at least one response")
+				return
+			}
+
+			var metricResp *MetricsResponse
+			for _, resp := range responses {
+				if resp.Key == tt.expectedMetricKey {
+					metricResp = &resp
+					break
+				}
+			}
+
+			if metricResp == nil {
+				t.Error("RPS metric not found in response")
+				return
+			}
+
+			if metricResp.Value != tt.expectedMetricValue {
```

Comment:
[optional] `require.Equal` or `require.EqualValues` can be used here to be more concise. I think there might be more occurrences in these tests where `require` can make this code easier to read.